src/instaskip/impl/from_eskip.clj

```diff
@@ -2,25 +2,26 @@
   (:require [instaparse.core :as insta]
             [clojure.data.json :as json]))
 
-(def ^:private eskip-routes-parser
+(def ^{:private true :testable true} eskip-routes-parser
   "returns an parser for eskip routes"
   (insta/parser
     "
      routes           = route+
      route            = name <':'> predicates <'->'> filters endpoint <';'>
-     name             = #'[a-zA-Z0-9]+'
+     name             = #'[a-zA-Z0-9_]+'
      filters          = (filter <'->'> )*
      predicates       = predicate (<'&&'> predicate)* | star
      predicate        = predicate-name <'('> predicate-arg? (<','> predicate-arg)* <')'>
      predicate-name   = #'[a-zA-Z0-9]*'
-     predicate-arg    = string | number | regexval
+     predicate-arg    = string-arg | number | regexval
      filter           = filter-name <'('> filter-arg? (<','> filter-arg)* <')'>
      filter-name      = #'[a-zA-Z0-9]*'
-     filter-arg       = string | number
+     filter-arg       = string-arg | number | regexval
      endpoint         = string | shunt
      shunt            = #'<shunt>'
      star             = #'\\*'
      string           = <quote> #'[^\"]*' <quote>
+     string-arg       = string
      quote            = #'\"'
      number           = decimal-number | int-number
      decimal-number   = #'[0-9]+\\.[0-9]*'
```

Comment:
How many different eskip grammars exist by now? at least four, right? The original go one, the haskell-parsec one, this one and there isn't there also one in scala?